compiler/tests/integration.bal

```diff
@@ -0,0 +1,21 @@
+import ballerina/test;
+import ballerina/file;
+
+final string SOURCE_DIR = check file:joinPath("..", "tests");
+
+@test:Config {
+    dataProvider: listPositive
+}
+function testPositive(string path) returns error? {
+    // JBUG due to #31146 let's assert so we see the file name
+    error? err = main(path);
```

Comment:
I don't like calling `main` here. Better to factor out the relevant part of main into a `compileFile` function and call that.